src/service/hooks.ts

```diff
@@ -0,0 +1,64 @@
+import { useState, useEffect, useRef } from 'react';
+import { Image } from './type';
+import axios from 'axios';
+
+export const useFetchData = (path: string) => {
+    const [data, setData] = useState<Image[]>([]);
+    const [loading, setLoading] = useState(false);
+    const [hasError, setHasError] = useState(false);
+
+    useEffect(() => {
+        if (path) {
+            const isSearch = path.includes('search');
+            setLoading(true);
+            setHasError(true);
```

Comment:
I believe this should be setHasError(false)
In browser the "Something goes wrong" is always shown in this case.